spikeextractors/extractors/shybridextractors/shybridextractors.py

```diff
@@ -0,0 +1,159 @@
+import os
+
+import numpy as np
+
+from spikeextractors import SortingExtractor
+from spikeextractors.extractors.bindatrecordingextractor import BinDatRecordingExtractor
+from spikeextractors.extraction_tools import save_to_probe_file
+
+try:
+    import hybridizer.io as sbio
+    import hybridizer.probes as sbprb
+    HAVE_SBEX = True
+except ImportError:
+    HAVE_SBEX = False
+
+class SHYBRIDRecordingExtractor(BinDatRecordingExtractor):
+    extractor_name = 'SHYBRIDRecordingExtractor'
+    installed = HAVE_SBEX
+    extractor_gui_params = [
+        {'name': 'file_path', 'type': 'file', 'title': "Full path to hybrid recording (.bin, .raw, .dat)"},
+    ]
+    is_writable = True
+
+    def __init__(self, file_path):
+        # load params file related to the given shybrid recording
+        params = sbio.get_params(file_path)['data']
+
+        # create a shybrid probe object
+        probe = sbprb.Probe(params['probe'])
+        nb_channels = probe.total_nb_channels
+
+        # translate the byte ordering
+        # TODO still ambiguous, shybrid should assume time_axis=1, since spike interface makes an assumption on the byte ordering
+        byte_order = params['order']
+        if byte_order == 'C':
+            time_axis = 1
+        elif byte_order == 'F':
+            time_axis = 0
+
+        # piggyback on binary data recording extractor
+        BinDatRecordingExtractor.__init__(self,
```

Comment:
would the broken channels be in the probe file? In that case, you could add this line after the init:

`self = se.load_probe_file(self, probe)`

This will return a SubRecordingExtractor with the channel map and your object will be set to this new extractor.
Locations will be also loaded automatically.
Does it make sense?

Reply:
The probe file indeed contains the information about the broken channels. As per your suggestion (https://github.com/jwouters91/shybrid/issues/1#issuecomment-536643086) I removed loading the probe file completely from the extractor. Now the user will have to explicitly load the probe file (https://github.com/jwouters91/shybrid/blob/f33e4fc732fcb3d6ca76182265c349ff229f4382/examples/spikeinterface_validation_example.py#L22)

Which style do you think is most in line with the framework?

Reply:
I guess I don't agree with my past self...I think at that time I didn't realize that the probe file is expected to be there in SHYBRID format. I think it would be nice to load it automatically. Sorry for my swinging opinions